content/6_Advanced/LineContainer.mdx

```diff
@@ -30,13 +30,161 @@ frequency: 1
 	</Resource>
 </Resources>
 
-<IncompleteSection>Example problem + implementation?</IncompleteSection>
+A half-plane can be defined as a planar region that consists of all points on one side of an infinite straight line, and none on the other side. We can apply computational geometry here, as an intersection can be represented as a convex polygon, where every point inside it is also inside all half-planes. If we can construct this polygon, we can retrieve the intersections.
 
-<Problems problems="half" />
+We can state that $N$ is the number of half-planes in our set. We represent lines and half-planes by a point and a vector, or in other terms, any point that lies on said line and the direction vector of the line. Each half-plane allows the region to the left of its direction vector. We can define the angle of a given half-plane as the polar coordinates of its direction vector. We can also state that a resulting intersection is always either bounded or empty, in cases where this doesn't hold, we can add 4 more half-planes which increase the size of the bounding box. For now, we shall also assume that parallel half-planes will not be present. To get a more visual representation of this, see the image below:
 
-## LineContainer (aka $\mathcal{O}(N \log N)$ CHT)
+![halfplane1](https://github.com/cpinitiative/usaco-guide/assets/145281030/bf54b84b-038f-47bd-a7e7-2f17ed8b8504)
 
-<Problems problems="sample" />
+This half-plane has an equation $y = 2x -2$ and can be represented as $P = (1,0)$ with the direction vector $PQ$ as $(1,2)$.
+I will now describe various ways of solving this problem, explaining algorithms in increasing efficiency:
```

Comment:
wait so is there a focus problem for this

Reply:
wait i just took a look at your code
there's no main method?

Reply:
it's in c++?

Reply:
no yeah but,, you know most of the code here is intended to be submitted to a problem for ac purposes
your code here has no main method, so

Reply:
this problem is not from any contests

Reply:
then how can we make sure it's correct

Reply:
the provided solution is correct. it does work.

Reply:
can you send a proof of ac on a problem that used it?

Reply:
sure, but this code does work for this problem, like official solution.
and acceptance for what test cases in what problem? there's none.
the example problem's in the module, for which this solution works.

Reply:
> this problem

what problem?

Reply:
which specific one?

Reply:
the one explained in this module with pictures

Reply:
unless you have an actual problem with test cases i'm pretty leery of adding this

Reply:
here, how about this: https://www.codechef.com/problems/CHN02.
i can also add extra practice problems on this topic.

Reply:
as long as the sol works on that

Reply:
proof that the solution works on this problem:
![image](https://github.com/cpinitiative/usaco-guide/assets/145281030/ea52b1c2-4830-4b1d-b577-6f3dbc57538d)

Reply:
also, my code does have a main method; i just looked.

Reply:
and once you merge this, i can add a bunch of practice problems i found (from actual contests with test cases) in the json file.

Reply:
> also, my code does have a main method; i just looked.

what line in the file?

> and once you merge this, i can add a bunch of practice problems i found (from actual contests with test cases) in the json file.

why not add them now?

Reply:
bcz rn the current module is unstructured without proper sections for the problems for each topic. once this is merged, the problems section will be formatted correctly.

Reply:
you can make the sections yourself my guy

Reply:
ik and i did in this pr. but idk exactly where the problems i add in the json file would show up in the current module bcz rn it has like 3 problem sections when it only needs 2 (one for half-plane, other for line-container). if this pr is merged, there will be 2 problem sections like there supposed to and then i can add the problems. look at the current module now and u'll see how disorganized it is.

Reply:
why...would merging this turn it back into 2 problem sections?

also
> what line in the file?

Reply:
nvm, i guess i'll still add those problems. and nvm for that too; i thought i saw main, but nope didn't

Reply:
@SansPapyrus683 i added those problems for half-plane intersection in the json file in this pr #4577